packages/ui/src/components/NumberInputV2/index.tsx

```diff
@@ -0,0 +1,424 @@
+import styled from '@emotion/styled'
+import { Icon } from '@ultraviolet/icons'
+import type {
+  ChangeEvent,
+  ForwardedRef,
+  InputHTMLAttributes,
+  ReactNode,
+} from 'react'
+import {
+  createRef,
+  forwardRef,
+  useCallback,
+  useId,
+  useImperativeHandle,
+  useMemo,
+} from 'react'
+import { Button } from '../Button'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+import { Tooltip } from '../Tooltip'
+
+const SIZES = {
+  small: '30px',
+  medium: '38px',
+  large: '46px',
+}
+
+type Sizes = keyof typeof SIZES
+
+const SideContainer = styled(Stack)`
+  padding: ${({ theme }) => `${theme.space['0.25']} ${theme.space['1']}`};
+
+  &[data-size='small'] {
+    height: ${SIZES.small};
+  }
+
+  &[data-size='medium'] {
+    height: ${SIZES.medium};
+  }
+
+  &[data-size='large'] {
+    height: ${SIZES.large};
+    padding: ${({ theme }) => `${theme.space['0.5']} ${theme.space['1']}`};
+  }
+`
+
+const InputContainer = styled(Stack)`
+  border-width: 0 1px 0 1px;
+  border-style: solid;
+  border-color: inherit;
+  background: inherit;
+`
+
+const Unit = styled(Text, {
+  shouldForwardProp: prop => !['size'].includes(prop),
+})<{ size: Sizes }>`
+  display: flex;
+  align-items: center;
+  padding: ${({ theme }) =>
```

Comment:
`padding: ${({ theme }) => theme.space['1]}` should have the same result since for left  (missing value) it will take  the first value (like a loop)